internal/provisioner/registry/main.go

```diff
@@ -0,0 +1,214 @@
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/url"
+	"os"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	crfinalizer "sigs.k8s.io/controller-runtime/pkg/finalizer"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+	"github.com/operator-framework/rukpak/internal/finalizer"
+	"github.com/operator-framework/rukpak/internal/provisioner/registry/controllers"
+	"github.com/operator-framework/rukpak/internal/source"
+	"github.com/operator-framework/rukpak/internal/storage"
+	"github.com/operator-framework/rukpak/internal/util"
+	"github.com/operator-framework/rukpak/internal/version"
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(apiextensionsv1.AddToScheme(scheme))
+	utilruntime.Must(rukpakv1alpha1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	var (
+		httpBindAddr         string
+		httpExternalAddr     string
+		enableLeaderElection bool
+		probeAddr            string
+		systemNamespace      string
+		unpackImage          string
+		rukpakVersion        bool
+		gitClientImage       string
+		storageDirectory     string
+	)
+	flag.StringVar(&httpBindAddr, "http-bind-address", ":8080", "The address the http server binds to.")
+	flag.StringVar(&httpExternalAddr, "http-external-address", "http://localhost:8080", "The external address at which the http server is reachable.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.StringVar(&systemNamespace, "system-namespace", "rukpak-system", "Configures the namespace that gets used to deploy system resources.")
+	flag.StringVar(&unpackImage, "unpack-image", "quay.io/operator-framework/rukpak:latest", "Configures the container image that gets used to unpack Bundle contents.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	flag.BoolVar(&rukpakVersion, "version", false, "Displays rukpak version information")
+	flag.StringVar(&storageDirectory, "storage-dir", storage.DefaultBundleCacheDir, "Configures the directory that is used to store Bundle contents.")
+	flag.StringVar(&gitClientImage, "git-client-image", "alpine/git:v2.32.0", "Configures which git container image to use to clone bundle git repos.")
```

Comment:
nit: do we need git support for the registry+v1 provisioner? Unless we are introducing supporting sourcing registry+v1 bundles from git as well? I don't think it's a bad idea, if we get it for free, but we are sort-of introducing a new format this way.

Reply:
I think this is might be fine to keep in right now but I do wonder whether we'd ever add support for sourcing from git repositories using this provisioner.

Reply:
In my mind, the method that we use to arrive at a `fs.FS` that represents the bundle is fully divorced from the contents and schema of that `fs.FS` (plain+v0, registry+v1, whatever)

IMO we should stop co-mingling the ideas as part of the rukpak migration. The conversation yesterday about sourcing from `ConfigMaps` would have the same question right? "Do we need ConfigMap support for the registry+v1 provisioner"

I can't think of a good reason to artificially limit how we unpack bundles because all of that unpack code is reused no matter what the format is.

Reply:
Yup, that's fair -- I think decoupling the two makes total sense.

I think my concern here was that down the line, in a downstream context, users may run into issues with sourcing registry+v1 bundles from git (not related to the implementation, but in general) and then we would be on the hook for supporting them. But I think the point about not limiting how we unpack bundles outweighs that concern.